x/pki/x509/x509.go

```diff
@@ -43,10 +45,24 @@ func DecodeX509Certificate(pemCertificate string) (*X509Certificate, error) {
 		return nil, types.NewErrInvalidCertificate(fmt.Sprintf("Could not parse certificate: %v", err.Error()))
 	}
 
+	oldVID := []byte("1.3.6.1.4.1.37244.2.1")
```

Comment:
Better to move the logic to a new function like:
`patchCertificate(ceritificate X509Certificate) X509Certificate`
which accepts the certificate as it is and patches the issuer and subject fields if applicable.
Consider also mentioning the ticket in a comment to explain the purpose of the function

Reply:
Done! ✅